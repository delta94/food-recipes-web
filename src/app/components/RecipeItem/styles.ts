import styled from 'styled-components'

export const Container = styled.main`
     margin-top: 1rem;
    display: flex;
    justify-content: space-between;
    align-items: center;

    h2 {
        cursor: pointer;
    }

    img {
        cursor: pointer;
        object-fit: cover;
        max-width: 100%;
        width: 100px;
        height: 100px;
        border-radius: 50%;
    }

    button {
        cursor: pointer;
        border: 0;
        border-radius: 25px;
        padding: 0.5rem 1.6rem;
        background-color: #f6e5c9;
        color: black;
  }
`
