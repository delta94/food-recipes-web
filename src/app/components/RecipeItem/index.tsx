import React, { MouseEvent, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { Creators } from '../../../core/redux/store/ducks/recipe'
import Modal from 'react-modal'
import Recipe from '../RecipeHome'
import { baseURLImages } from '../../../config/url'

import { Container } from './styles'
import UpRecipe from '../UpRecipe'

interface IRecipeItem {
  id: string
  name: string
  image: string
  ingredients: string
  mode_prepare: string
  time: string
}

const RecipeItem = ({
  id,
  name,
  image,
  ingredients,
  mode_prepare,
  time
}: IRecipeItem) => {
  const [modalRecipeViewIsOpen, setModalRecipeViewIsOpen] = React.useState(
    false
  )
  const [modalRecipeEditIsOpen, setModalRecipeEditIsOpen] = React.useState(
    false
  )

  const dispatch = useDispatch()

  function handleDelete (event: MouseEvent<HTMLButtonElement>) {
    event.preventDefault()

    dispatch(Creators.deleteRecipeRequest(id))
  }

  useEffect(() => {
    Modal.setAppElement('body')
  }, [])

  function openRecipeViewModal () {
    setModalRecipeViewIsOpen(true)
  }

  function closeRecipeViewModal () {
    setModalRecipeViewIsOpen(false)
  }

  function openRecipeEditModal () {
    setModalRecipeEditIsOpen(true)
  }

  function closeRecipeEditModal () {
    setModalRecipeEditIsOpen(false)
  }

  const customStyles = {
    content: {
      marginLeft: 'auto',
      marginRight: 'auto',
      maxWidth: '1000px'
    }
  }

  return (
    <Container>
      <img
        onClick={openRecipeViewModal}
        src={`${baseURLImages}/${image}`}
        alt={name}
      />
      <Modal
        isOpen={modalRecipeViewIsOpen}
        onRequestClose={closeRecipeViewModal}
        style={customStyles}
      >
        <Recipe
          id={id}
          name={name}
          image={image}
          mode_prepare={mode_prepare}
          ingredients={ingredients}
          time={time}
        />
      </Modal>
      <h2 onClick={openRecipeViewModal}>{name}</h2>
      <button onClick={openRecipeEditModal} type='submit'>
        Editar
      </button>
      <Modal
        isOpen={modalRecipeEditIsOpen}
        onRequestClose={closeRecipeEditModal}
        style={customStyles}
      >
        <UpRecipe
          idFrom={id}
          nameFrom={name}
          imageFrom={image}
          mode_prepareFrom={mode_prepare}
          ingredientsFrom={ingredients}
          timeFrom={time}
        />
      </Modal>
      <Link to='/'>
        <button type='submit' onClick={handleDelete}>
          Apagar
        </button>
      </Link>
    </Container>
  )
}

export default RecipeItem
