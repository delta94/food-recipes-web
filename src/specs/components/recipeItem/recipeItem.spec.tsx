import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import RecipeItem from '../../../app/components/RecipeItem'

import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

const mockStore = configureStore([])

describe('[COMPONENTS] recipeItem', () => {
  let store: any

  beforeEach(() => {
    store = mockStore({
      auth: {}
    })
  })

  it('Should render correctly', () => {
    const container = render(
      <BrowserRouter>
        <Provider store={store}>
          <RecipeItem
            id='any_id'
            name='any_name'
            image='any_image'
            ingredients='any_ingredients'
            mode_prepare='any_mode_prepare'
            time='any_time'
          />
        </Provider>
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('Should call function openRecipeViewModal on click modal', async () => {
    const { getByRole } = render(
      <BrowserRouter>
        <Provider store={store}>
          <RecipeItem
            id='any_id'
            name='any_name'
            image='any_image'
            ingredients='any_ingredients'
            mode_prepare='any_mode_prepare'
            time='any_time'
          />
        </Provider>
      </BrowserRouter>
    )

    expect(fireEvent.click(getByRole('img'))).toBeTruthy()
  })

  it('Should call function openRecipeEditModal on click button edit', async () => {
    const { getByText } = render(
      <BrowserRouter>
        <Provider store={store}>
          <RecipeItem
            id='any_id'
            name='any_name'
            image='any_image'
            ingredients='any_ingredients'
            mode_prepare='any_mode_prepare'
            time='any_time'
          />
        </Provider>
      </BrowserRouter>
    )

    expect(fireEvent.click(getByText('Editar'))).toBeTruthy()
  })

  it('Should call function handleDelete on click button delete', async () => {
    const { getByText } = render(
      <BrowserRouter>
        <Provider store={store}>
          <RecipeItem
            id='any_id'
            name='any_name'
            image='any_image'
            ingredients='any_ingredients'
            mode_prepare='any_mode_prepare'
            time='any_time'
          />
        </Provider>
      </BrowserRouter>
    )

    expect(fireEvent.click(getByText('Apagar')))
  })

  // it('Should call function closeRecipeViewModal  on click modal', async () => {
  //   const { getByRole, getByText } = render(
  //     <BrowserRouter>
  //       <Provider store={store}>
  //         <RecipeItem
  //           id='any_id'
  //           name='any_name'
  //           image='any_image'
  //           ingredients='any_ingredients'
  //           mode_prepare='any_mode_prepare'
  //           time='any_time'
  //         />
  //       </Provider>
  //     </BrowserRouter>
  //   )

  //   fireEvent.click(getByRole('img'))

  //   expect(fireEvent.click(document)).toBeTruthy()
  // })
})
