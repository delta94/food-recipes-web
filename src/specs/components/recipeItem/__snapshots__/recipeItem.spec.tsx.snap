// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] recipeItem Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-dlnjPT eCboIE"
      >
        <img
          alt="any_name"
          src="http://localhost:3333/files/any_image"
        />
        <h2>
          any_name
        </h2>
        <button
          type="submit"
        >
          Editar
        </button>
        <a
          href="/"
        >
          <button
            type="submit"
          >
            Apagar
          </button>
        </a>
      </main>
    </div>
    <div
      class="ReactModalPortal"
    />
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <main
      class="sc-dlnjPT eCboIE"
    >
      <img
        alt="any_name"
        src="http://localhost:3333/files/any_image"
      />
      <h2>
        any_name
      </h2>
      <button
        type="submit"
      >
        Editar
      </button>
      <a
        href="/"
      >
        <button
          type="submit"
        >
          Apagar
        </button>
      </a>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
